specs.bzl

```diff
@@ -246,14 +246,21 @@ def _artifact_to_coord(artifact):
     classifier = (",classifier=" + artifact["classifier"]) if artifact.get("classifier") != None else ""
     return artifact["group"] + ":" + artifact["artifact"] + ":" + artifact["version"] + classifier
 
-def _repository_url(repository_spec):
-    (protocol, remainder) = repository_spec["repo_url"].split("//")
+#
+# Returns a string "{hostname} {user}:{password}" for a repository_spec
+#
+def _repository_credentials(repository_spec):
+    if "credentials" not in repository_spec:
+        fail("Asked to generate credentials for a repository that does not need it")
+
+    (_, remainder) = repository_spec["repo_url"].split("//")
+    hostname = remainder.split("/")[0]
+
+    credentials = repository_spec["credentials"]
 
-    credentials = repository_spec.get("credentials")
-    auth_string = (credentials["user"] + ":" + credentials["password"] + "@") if credentials != None else ""
-    return protocol + "//" + auth_string + remainder
+    return hostname + " " + credentials["user"] + ":" + credentials["password"]
```

Comment:
Please add a comment linking to https://get-coursier.io/docs/other-credentials.html#inline to have the context for this format